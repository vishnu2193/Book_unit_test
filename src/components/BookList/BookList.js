import React, { Component } from "react";
import Book from "../Book/Book";
export default class BookList extends Component {
  render() {
    return (
      <div>
        <h1>Welcome to My Favorite Book Portal</h1>
        <Book name="Physics" />
      </div>
    );
  }
}
