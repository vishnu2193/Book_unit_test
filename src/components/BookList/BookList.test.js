import BookList from "./BookList";
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";

describe("BookListComponent", () => {
  //# 3 Have a right Book­List com­po­nent snap­shot
  it("has right booklist component snapshop", () => {
    const snapshotJSON = renderer.create(<BookList />);
    expect(snapshotJSON).toMatchSnapshot();
  });

  // #4 Test whether Book­List com­po­nent has the fol­low­ing text in it. ‘Wel­come to My Fa­vorite Book Por­tal’
  it("has Welcome to My Favorite Book Portal!", () => {
    render(<BookList />);
    const welcomeText = screen.getByText("Welcome to My Favorite Book Portal");
    expect(welcomeText).toBeInTheDocument();
  });
});
