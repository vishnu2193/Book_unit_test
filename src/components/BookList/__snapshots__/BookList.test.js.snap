// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookListComponent has right booklist component snapshop 1`] = `
<div>
  <h1>
    Welcome to My Favorite Book Portal
  </h1>
  <div>
    <div
      data-testid="bookName"
    >
      Physics
    </div>
    <button
      data-testid="addButton"
      onClick={[Function]}
    >
      Add to Product List
    </button>
  </div>
</div>
`;
